mindsdb_native/libs/phases/data_extractor/data_extractor.py

```diff
@@ -88,6 +91,43 @@ def _get_prepared_input_df(self):
                 # if no data frame yet, make one
                 df = self._data_from_when()
 
+            if self.transaction.lmd['setup_args'] is not None and self.transaction.lmd['tss']['is_timeseries']:
+                if 'make_predictions' not in df.columns:
+                    df['make_predictions'] = [True] * len(df)
+
+                setup_args = deepcopy(self.transaction.lmd['setup_args'])
+
+
+                if len(df) > 1 and self.transaction.lmd['tss']['group_by'] is not None:
+                    encountered_set = set()
+                    unique_group_by_rows = []
+                    for i in range(en(df)):
+                        val_str = ''
+                        for group_col in self.transaction.lmd['tss']['group_by']:
+                            val_str += '###-------------###' + str(df.iloc[i][group_col])
+                            if val_str not in encountered_set:
+                                encountered_set.add(val_str)
+                                unique_group_by_rows.append(df.iloc[i])
+                else:
+                    unique_group_by_rows = [df.iloc[0]]
```

Comment:
`len(df)` is equal to `0` in this `else` , so `df.iloc[0]` is an `IndexError`

Reply:
`len(df) > 1` so length should be 1, the dataframe should never be 0